### ISSUE TITLE
Dotnet-Support

### ISSUE THREAD

**Issue by ckotzbauer** [anchore/grype#736]
**What would you like to be added**:
Support for .NET Core (relates to anchore/syft#951)

**Why is this needed**:
Dotnet is an important tech-ecosystem.

**Additional context**:
I tried to add support by myself, but it didn't work with a prepared docker-image (or SBOM from the image)
1. Updated syft-dependency to my local fork from anchore/syft#951
2. Implemented and registered a matcher for dotnet (like the python-matcher)
3. Grype didn't find vulnerable dotnet-packages from my SBOM (they were present in the Grype-DB).

Did I miss something here?

**Comment by olljanat** [comment]
Perhaps you need enable `RestorePackagesWithLockFile` setting ( https://devblogs.microsoft.com/nuget/enable-repeatable-package-restores-using-a-lock-file/ ) on projects which you are scanning like it is needed on Trivy?

**Comment by ckotzbauer** [comment]
I don't think so @olljanat. The packages are listed in the SBOM, so it's no MS related problem.
From anchore/syft#951:
> Looks like there are minor tweaks in grype testing needed -- I can confirm there is no apparent unexpected matching behavior with the stock matcher. We'll still need to add a .NET-specific matcher and DB namespace finder to link up GHSA matches in the future.

**Comment by olljanat** [comment]
@ckotzbauer Do I read this right that your implementation only do scan vulnerabilities in packages which are used by .NET project? Not vulnerabilities in .NET it selves? Tested with `grype mcr.microsoft.com/dotnet/runtime:6.0.5` and it does not list *CVE-2022-30184* which [was fixed by 6.0.6](https://github.com/dotnet/core/blob/main/release-notes/6.0/6.0.6/6.0.6.md)

**Comment by ckotzbauer** [comment]
Yes, currently only project dependencies are parsed.

**Comment by ckotzbauer** [comment]
@olljanat I checked this again: The built-in packages are listed and scanned correctly from my dotnet-code in syft and grype:
```bash
grype mcr.microsoft.com/dotnet/sdk:6.0-alpine  
 ✔ Vulnerability DB        [no update available]
New version of grype is available: 0.41.0 (currently running: 0.40.0)
 ✔ Loaded image            
 ✔ Parsed image            
 ✔ Cataloged packages      [1666 packages]
 ✔ Scanned image           [20 vulnerabilities]
NAME                            INSTALLED   FIXED-IN   TYPE    VULNERABILITY        SEVERITY 
Newtonsoft.Json                 9.0.1       13.0.1     dotnet  GHSA-5crp-9r3c-p9vr  High      
NuGet.CommandLine.XPlat         6.2.1-rc.2  6.2.1      dotnet  GHSA-3885-8gqc-3wpf  Medium    
NuGet.Commands                  6.2.1-rc.2  6.2.1      dotnet  GHSA-3885-8gqc-3wpf  Medium    
System.Net.Http                 4.3.0       4.3.4      dotnet  GHSA-7jgj-8wvc-jh57  High      
System.Text.RegularExpressions  4.3.0       4.3.1      dotnet  GHSA-cmhx-cq75-c4mj  Medium    
libcrypto1.1                    1.1.1o-r0   1.1.1q-r0  apk     CVE-2022-2097        Unknown   
libssl1.1                       1.1.1o-r0   1.1.1q-r0  apk     CVE-2022-2097        Unknown
```

The reason why the Nuget.* deps are not listed for the runtime-image is, they are not included there (as nuget is used at compile-time and not at runtime).

So there should be no issue with the scanning here.

**Comment by olljanat** [comment]
@ckotzbauer that explains the result. Thanks for sharing.
